### PR TITLE
Fix nested DelegatingLogger issue.

### PR DESCRIPTION
**Pull request by benbelow**
Ensures only one layer of DelegatingLogger is created. Currently if you run a renderer with logging, you end up with many nested loggers which slows performance and eventually crashes the renderer. It would be useful to us at Softwire if this could be merged soon.

### CODE REVIEW

File: lib/alephant/logger.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def self.get_logger
     end
 
     def self.set_logger(value)
-      @@logger = Alephant::DelegatingLogger.new value
+      @@logger = Alephant::DelegatingLogger.new value if !@@logger.is_a?(DelegatingLogger)
```
Comment (revett) [inline lib/alephant/logger.rb]: You can use this instead:

```ruby
@@logger ||= Alephant::DelegatingLogger.new value
```

**Comment by revett** [comment]
Could you give an example of where this error occurs please?

File: lib/alephant/logger.rb (hunk 1 above)
Comment (benbelow) [inline lib/alephant/logger.rb]: You're right, I'll change that.

**Comment by benbelow** [comment]
We're using the logger in conjunction with Alephant-Publisher-Queue, which uses Alephant-Lookup. 
The writer creates a LookupHelper, which calls Logger.set_logger in its constructor. Without the check to see if it has already been initialized, this creates a new DelegatingLogger wrapping the currently existing DelegatingLogger.

This means that our renderer seems to be making a new DelegatingLogger for each item it renders, which gradually slows down the renderer until it crashes after a few hours.

**Comment by revett** [comment]
Your change would mean that the `set_logger` method will only actually set the logger on it's first call (not inline with expected behaviour). The problem is instead within `LookupHelper` as you pointed out.

**Comment by benbelow** [comment]
The problem will still occur any time set_logger is called with a DelegatingLogger, such as that returned by get_logger. The latest change I've pushed should be able to prevent that by only wrapping the value in a DL if it is not already a DL, while still allowing set_logger to set the logger more than once.

**Comment by benbelow** [comment]
I have also submitted a PR to fix the misuse of the logger in the LookupHelper : https://github.com/BBC-News/alephant-lookup/pull/5

**Comment by Integralist** [comment]
Although this seems like a small PR, it has actually opened up a much bigger discussion amongst the @BBC-News/framework team. There is fundamental issues with the original design of Alephant-Logger.

@revett and @darnould will be discussing this internally, and then will share thoughts here.

**Comment by Integralist** [comment]
I'm going to close this PR, as we've now re-architectured the Logger functionality within Alephant:

- [New Alephant-Logger example](https://github.com/BBC-News/alephant-logger/blob/master/README.md#usage)

The example demonstrates using two *driver* gems we've released to accommodate different functionality. The drivers are passed into Alephant Logger, and we include the standard Ruby Logger as well.

**Pull request closed without merging**